tsfile/src/test/java/org/apache/iotdb/tsfile/read/ReadBugTest.java

```diff
@@ -0,0 +1,208 @@
+
+    rowCount = 1000;
+    chunkGroupSize = 16 * 1024 * 1024;
+    pageSize = 10000;
+    generateSampleInputDataFile();
+    writeToFile();
+
+    TsFileSequenceReader reader = new TsFileSequenceReader(outputDataFile);
+    roTsFile = new ReadOnlyTsFile(reader);
+  }
+
+  private void generateSampleInputDataFile() throws IOException {
+    File file = new File(inputDataFile);
+    if (file.exists()) {
+      file.delete();
+    }
+    file.getParentFile().mkdirs();
+    FileWriter fw = new FileWriter(file);
+
+    long startTime = START_TIMESTAMP;
+    for (int i = 0; i < rowCount; i++) {
+      // write d1
+      String d1 = "d1," + (startTime + i);
+      if (i % 8 == 0) {
+        d1 += ",s4," + "dog" + i;
+      }
+      fw.write(d1 + "\r\n");
+    }
+    fw.close();
+  }
+
+  private void writeToFile() throws IOException, WriteProcessException {
+    File file = new File(outputDataFile);
+    if (file.exists()) {
+      file.delete();
+    }
+
+    FileSchema schema = generateTestSchema();
+
+    TSFileDescriptor.getInstance().getConfig().groupSizeInByte = chunkGroupSize;
+    TSFileDescriptor.getInstance().getConfig().maxNumberOfPointsInPage = pageSize;
+    TsFileWriter innerWriter = new TsFileWriter(file, schema, TSFileDescriptor.getInstance().getConfig());
+    Scanner in = getDataFile(inputDataFile);
+    assert in != null;
+    while (in.hasNextLine()) {
+      String str = in.nextLine();
+      TSRecord record = RecordUtils.parseSimpleTupleRecord(str, schema);
+      innerWriter.write(record);
+    }
+    innerWriter.close();
+    in.close();
+  }
+
+  private Scanner getDataFile(String path) {
+    File file = new File(path);
+    try {
+      Scanner in = new Scanner(file);
+      return in;
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
+      return null;
+    }
+  }
+
+  private FileSchema generateTestSchema() {
+    SchemaBuilder schemaBuilder = new SchemaBuilder();
+//    schemaBuilder.addSeries("s1", TSDataType.INT32, TSEncoding.RLE);
+//    schemaBuilder.addSeries("s2", TSDataType.INT64, TSEncoding.PLAIN);
+//    schemaBuilder.addSeries("s3", TSDataType.INT64, TSEncoding.TS_2DIFF);
+    schemaBuilder.addSeries("s4", TSDataType.TEXT, TSEncoding.PLAIN, CompressionType.UNCOMPRESSED,
+        Collections.singletonMap(Encoder.MAX_STRING_LENGTH, "20"));
+//    schemaBuilder.addSeries("s5", TSDataType.BOOLEAN, TSEncoding.RLE);
+//    schemaBuilder.addSeries("s6", TSDataType.FLOAT, TSEncoding.RLE, CompressionType.SNAPPY,
+//        Collections.singletonMap(Encoder.MAX_POINT_NUMBER, "5"));
+//    schemaBuilder.addSeries("s7", TSDataType.DOUBLE, TSEncoding.GORILLA);
+    return schemaBuilder.build();
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    if (roTsFile != null) {
+      roTsFile.close();
+    }
+
+    File file = new File(inputDataFile);
+    if (file.exists()) {
+      file.delete();
+    }
+    file = new File(outputDataFile);
+    if (file.exists()) {
+      file.delete();
+    }
```

Comment:
Have you tried on windows platform that these files can be truly deleted ?
Note that if you only test on mac/linux, you can not count on that windows have same result.

Reply:
use `Assert.assertTrue(file.delete())` ;